src/tuner/nccl_ofi_tuner.c

```diff
@@ -13,404 +13,145 @@
-					      (nccl_ofi_tuner_point_t){TUNER_MAX_SIZE, TUNER_MAX_RANKS});
-
-			const nccl_ofi_tuner_region_t regions[] = {
-				{.algorithm = NCCL_ALGO_TREE,
-				 .protocol = NCCL_PROTO_LL128,
-				 .num_vertices = 5,
-				 .vertices = {{0, 16}, {2367488, 16}, {9999360, 64}, {119477248, 128}, extended_tree_ll128}},
-				{.algorithm = NCCL_ALGO_RING,
-				 .protocol = NCCL_PROTO_LL128,
-				 .num_vertices = 9,
-				 .vertices = {{0, 2},
-					      {4736000, 2},
-					      {269484032, 128},
-					      extended_ring_ll128,
-					      extended_tree_ll128,
-					      {119477248, 128},
-					      {9999360, 64},
-					      {2367488, 16},
-					      {0, 16}}},
-				{.algorithm = NCCL_ALGO_RING,
-				 .protocol = NCCL_PROTO_SIMPLE,
-				 .num_vertices = 4,
-				 .vertices = {{4736000, 2}, {TUNER_MAX_SIZE, 2}, extended_ring_ll128, {269484032, 128}}}};
-
-			ret = set_regions(nccl_ofi_tuner_ctx, collType, sizeof(regions) / sizeof(regions[0]), regions);
-			if (ret != ncclSuccess) {
-				goto exit;
-			}
-		}
-		{
-			collType = ncclFuncAllGather;
-			nccl_ofi_tuner_point_t extended_ring_simple =
-				extend_region((nccl_ofi_tuner_point_t){4194304, 2},
-					      (nccl_ofi_tuner_point_t){8589934592, 2048},
-					      (nccl_ofi_tuner_point_t){TUNER_MAX_SIZE, TUNER_MAX_RANKS});
-
-			const nccl_ofi_tuner_region_t regions[] = {{.algorithm = NCCL_ALGO_RING,
-								    .protocol = NCCL_PROTO_SIMPLE,
-								    .num_vertices = 4,
-								    .vertices = {{4194304, 2},
-										 {TUNER_MAX_SIZE, 2},
-										 extended_ring_simple,
-										 {8589934592, 2048}}}};
-
-			ret = set_regions(nccl_ofi_tuner_ctx, collType, sizeof(regions) / sizeof(regions[0]), regions);
-			if (ret != ncclSuccess) {
-				goto exit;
-			}
-		}
-		{
-			collType = ncclFuncReduceScatter;
-			nccl_ofi_tuner_point_t extended_ring_simple =
-				extend_region((nccl_ofi_tuner_point_t){8388608, 2},
-					      (nccl_ofi_tuner_point_t){4294967296, 1024},
-					      (nccl_ofi_tuner_point_t){TUNER_MAX_SIZE, TUNER_MAX_RANKS});
-
-			const nccl_ofi_tuner_region_t regions[] = {{.algorithm = NCCL_ALGO_RING,
-								    .protocol = NCCL_PROTO_SIMPLE,
-								    .num_vertices = 4,
-								    .vertices = {{8388608, 2},
-										 {TUNER_MAX_SIZE, 2},
-										 extended_ring_simple,
-										 {4294967296, 1024}}}};
-
-			ret = set_regions(nccl_ofi_tuner_ctx, collType, sizeof(regions) / sizeof(regions[0]), regions);
-			if (ret != ncclSuccess) {
-				goto exit;
-			}
-		}
+	/*
+	 * We reach here. It means the folowing two conditions are met.
+	 *  - "Internal" force is not set by env variable
+	 *  - at least one of "Region" or "Model" tuner is supported for the given platform, nRanks and nNodes
+	 */
+
+	/*
+	 * We choose "Region" over "Model" when both are supported.
+	 * TUNER_FORCE_TYPE env variable is ignored if the forced tuner type is not
+	 * supported by the given platform, nRanks and nNodes.
+	 */
+
+	if (region_support && !(model_support && is_force_type_model)) {
```

Comment:
This works for this PR, but noting I can see a future where when a platform has a region defined and a model available... we can use the region till the scale at which it has empirical data for, and instead of extrapolating the edge polygons to infinity we would fallback to the model.

Reply:
We can support those hybrid case by return 1 or 0 based on "nRanks" and "nNodes" at `is_region_supported()` function. For now, in this PR, `is_region_supported()` returns 1 or 0 solely based on platform name.